### PR TITLE
Major update (v5) with a lot of different changes

### PR DESCRIPTION
**Pull request by tarampampam**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes

### Changed

- Maximal `illuminate/*` packages version now is `~6.0`
- Minimal `illuminate/*` packages version now is `^5.5`
- Minimal version of `avto-dev/extended-laravel-validator` package now `^3.2`
- Minimal version of `avto-dev/static-references-laravel` package now `^4.0`
- Typed IDEntities validation depends on `avto-dev/static-references-laravel` package service-provider loading
- `styleci.io` rules
- Config file location `./src/config/identity.php` &rarr; `./config/identity.php`
- Method signatures in `AvtoDev\IDEntity\IDEntity` class:
  - `::typeIsSupported($type): bool` &rarr; `::typeIsSupported(string $type): bool`
  - `::is(string $value, $type): bool` &rarr; `::is(string $value, string $type): bool`
- Method signatures in `AvtoDev\IDEntity\IDEntityInterface` interface:
  - `::is(string $value, $type): bool` &rarr; `::is(string $value, string $type): bool`
- Constructor in `AvtoDev\IDEntity\Types\AbstractTypedIDEntity` finalized
- Method signatures in `AvtoDev\IDEntity\HasCadastralNumberInterface` interface:
  - `->getRegionData(): ?\AvtoDev\StaticReferences\References\CadastralDistricts\CadastralRegionEntry` &rarr; `->getDistrictData(): ?\AvtoDev\StaticReferences\References\Entities\CadastralDistrict`
- Method signatures in `AvtoDev\IDEntity\HasRegionDataInterface` interface:
  - `->getRegionData(): ?\AvtoDev\StaticReferences\References\AutoRegions\AutoRegionEntry` &rarr; `->getRegionData(): ?\AvtoDev\StaticReferences\References\Entities\SubjectCodesInfo`
- Typed IDEntity classes now contains finalized static method `::make(string $value, ?string $type = null): self`
- Validation logic in `AvtoDev\IDEntity\Types\IDEntityCadastralNumber` now more strict
- Dev-dependencies `phpstan/phpstan` and `phpunit/phpunit` updated
- Unit-tests re-wrote

### Added

- GitHub Actions for a tests running
- PHP 7.4 support
- `tarampampam/wrappers-php` dependency
- Class `AvtoDev\IDEntity\Types\IDEntityCadastralNumber` now contains new methods:
  - `->getDistrictCode(): ?int`
  - `->getAreaCode(): ?int`
  - `->getSectionCode(): ?int`
  - `->getParcelCode(): ?int`

### Removed

- Dev-dependency `mockery/mockery`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code
- [x] I have made changes in [CHANGELOG.md](https://github.com/avto-dev/identity-laravel/blob/master/CHANGELOG.md) file

### CODE REVIEW

File: src/Types/IDEntityBody.php [hunk 1]
```diff
@@ -55,9 +63,13 @@ public static function normalize($value): ?string
      */
     public function isValid(): bool
     {
-        /** @var BodyCodeValidatorExtension $validator */
-        $validator = static::getContainer()->make(BodyCodeValidatorExtension::class);
+        if (\is_string($this->value) && $this->value !== '') {
+            /** @var BodyCodeValidatorExtension $validator */
+            $validator = static::getContainer()->make(BodyCodeValidatorExtension::class);
+
+            return \is_string($this->value) && $validator->passes('', $this->value);
```
Comment (Reallife) [inline src/Types/IDEntityBody.php]: Redundant `\is_string($this->value)`. We are check this condition above.

File: src/Types/IDEntityChassis.php [hunk 2]
```diff
@@ -52,9 +60,13 @@ public static function normalize($value): ?string
      */
     public function isValid(): bool
     {
-        /** @var ChassisCodeValidatorExtension $validator */
-        $validator = static::getContainer()->make(ChassisCodeValidatorExtension::class);
+        if (\is_string($this->value) && $this->value !== '') {
+            /** @var ChassisCodeValidatorExtension $validator */
+            $validator = static::getContainer()->make(ChassisCodeValidatorExtension::class);
+
+            return \is_string($this->value) && $validator->passes('', $this->value);
```
Comment (Reallife) [inline src/Types/IDEntityChassis.php]: Redundant `\is_string($this->value)`. We are check this condition above.

File: src/Types/IDEntityPts.php [hunk 3]
```diff
@@ -45,9 +53,13 @@ public static function normalize($value): ?string
      */
     public function isValid(): bool
     {
-        /** @var PtsCodeValidatorExtension $validator */
-        $validator = static::getContainer()->make(PtsCodeValidatorExtension::class);
+        if (\is_string($this->value) && $this->value !== '') {
+            /** @var PtsCodeValidatorExtension $validator */
+            $validator = static::getContainer()->make(PtsCodeValidatorExtension::class);
+
+            return \is_string($this->value) && $validator->passes('', $this->value);
```
Comment (Reallife) [inline src/Types/IDEntityPts.php]: Redundant `\is_string($this->value)`. We are check this condition above.

File: src/Types/IDEntitySts.php [hunk 4]
```diff
@@ -25,17 +33,17 @@ public function getType(): string
...
-            $value = Str::upper(\trim((string) $value));
+            // Uppercase + trim
+            $value = \mb_strtoupper(\trim((string) $value), 'UTF-8');
 
-            // Удаляем все символы, кроме разрешенных
+            // Remove all chars except allowed
             $value = (string) \preg_replace('~[^' . 'АБВГДЕЖЗИКЛМНОПРСТУФХЦЧШЩЫЭЮЯ' . 'A-Z' . '0-9]~u', '', $value);
 
-            // Производим замену латинских аналогов на кириллические (обратная транслитерация)
+            // Replace latin- chars with kyr- analogs (backward transliteration)
```
Comment (Reallife) [inline src/Types/IDEntitySts.php]: Not important. Maybe change all `kyr` strings in comment and names of vars and constants to `cyrillic`? Diffucult for understand what `kyr` means.

File: src/Types/IDEntitySts.php [hunk 5]
```diff
@@ -45,9 +53,13 @@ public static function normalize($value): ?string
      */
     public function isValid(): bool
     {
-        /** @var StsCodeValidatorExtension $validator */
-        $validator = static::getContainer()->make(StsCodeValidatorExtension::class);
+        if (\is_string($this->value) && $this->value !== '') {
+            /** @var StsCodeValidatorExtension $validator */
+            $validator = static::getContainer()->make(StsCodeValidatorExtension::class);
+
+            return \is_string($this->value) && $validator->passes('', $this->value);
```
Comment (Reallife) [inline src/Types/IDEntitySts.php]: Same. Redundant `\is_string($this->value)`

File: src/Types/IDEntityVin.php [hunk 6]
```diff
@@ -85,9 +97,13 @@ public function isChecksumValidated(): bool
      */
     public function isValid(): bool
     {
-        /** @var VinCodeValidatorExtension $validator */
-        $validator = static::getContainer()->make(VinCodeValidatorExtension::class);
+        if (\is_string($this->value) && $this->value !== '') {
+            /** @var VinCodeValidatorExtension $validator */
+            $validator = static::getContainer()->make(VinCodeValidatorExtension::class);
+
+            return \is_string($this->value) && $validator->passes('', $this->value);
```
Comment (Reallife) [inline src/Types/IDEntityVin.php]: Same. Redundant `\is_string($this->value)`

File: src/Types/AbstractTypedIDEntity.php [hunk 7]
```diff
@@ -4,31 +4,30 @@
 
 namespace AvtoDev\IDEntity\Types;
 
+use function mb_strlen;
+use function mb_substr;
```
Comment (eldario) [inline src/Types/AbstractTypedIDEntity.php:8]: Maybe just from the `\`?
`\mb_strlen`
`\mb_substr`

File: src/Types/AbstractTypedIDEntity.php [hunk 8]
```diff
@@ -111,7 +104,7 @@ public function toArray(): array
      */
     public function toJson($options = 0): string
     {
-        return (string) \json_encode($this->toArray(), $options);
+        return Json::encode((object) $this->toArray(), $options);
```
Comment (eldario) [inline src/Types/AbstractTypedIDEntity.php]: > cast to `object`

What is this for?

File: src/Types/HasCadastralNumberInterface.php [hunk 9]
```diff
@@ -2,14 +2,14 @@
...
 namespace AvtoDev\IDEntity\Types;
 
-use AvtoDev\StaticReferences\References\CadastralDistricts\CadastralRegionEntry;
+use AvtoDev\StaticReferences\References\Entities\CadastralDistrict;
 
 interface HasCadastralNumberInterface
 {
     /**
-     * Return cadastral data by region & district codes.
+     * Get cadastral data by region & district codes.
```
Comment (eldario) [inline src/Types/HasCadastralNumberInterface.php]: Get cadastral data by **district** & **area** codes.

File: src/Types/AbstractTypedIDEntity.php (hunk 7 above)
Comment (tarampampam) [reply src/Types/AbstractTypedIDEntity.php:8]: There are a lot of this functions calling in a class. So, `use function` is a better choice for clean code, I think

File: src/Types/IDEntityBody.php (hunk 1 above)
Comment (tarampampam) [reply src/Types/IDEntityBody.php]: Yes, you right!

File: src/Types/IDEntityChassis.php (hunk 2 above)
Comment (tarampampam) [reply src/Types/IDEntityChassis.php]: Fixed

File: src/Types/IDEntityPts.php (hunk 3 above)
Comment (tarampampam) [reply src/Types/IDEntityPts.php]: Fixed

File: src/Types/IDEntitySts.php (hunk 4 above)
Comment (tarampampam) [reply src/Types/IDEntitySts.php]: Yeah, my bad

File: src/Types/IDEntitySts.php (hunk 5 above)
Comment (tarampampam) [reply src/Types/IDEntitySts.php]: Done

File: src/Types/IDEntityVin.php (hunk 6 above)
Comment (tarampampam) [reply src/Types/IDEntityVin.php]: Also fixed

File: src/Types/AbstractTypedIDEntity.php (hunk 8 above)
Comment (tarampampam) [reply src/Types/AbstractTypedIDEntity.php]: In json context this is not required, you right. FIxed

File: src/Types/IDEntityBody.php (hunk 1 above)
Comment (eldario) [inline src/Types/IDEntityBody.php]: Why do we check that this `\is_string` twice?

File: src/Types/HasCadastralNumberInterface.php (hunk 9 above)
Comment (tarampampam) [reply src/Types/HasCadastralNumberInterface.php]: Done

File: src/Types/IDEntityBody.php (hunk 1 above)
Comment (tarampampam) [reply src/Types/IDEntityBody.php]: Already removed, thx

**Review by Reallife — APPROVED** [review]

**Pull request merged**